v2/tools/generator/internal/astmodel/type_name_test.go

```diff
@@ -87,3 +87,120 @@ func TestTypeName_IsEmpty(t *testing.T) {
 	g.Expect(blank.IsEmpty()).To(BeTrue())
 	g.Expect(name.IsEmpty()).To(BeFalse())
 }
+
+type mockPackageReference struct {
```

Comment:
The interface `PackageReference` is intended to have a closed set of implementations - `LocalPackageReference`, `StoragePackageReference` and `ExternalPackageReference` only. This knowledge is baked in across `astmodel` in a bunch of places.

Introducing a new implementation, even if only for testing, introduces significant risk because things _will_ go :boom: as soon as it's used outside the context of this one file.

Instead, use the existing function `makeTestLocalPackageReference()` (see `local_package_reference_test.go`) to create your test references.

I'd also suggest factoring out the common package references as some local variables like this:

``` go
pkgv2 := makeTestLocalPackageReference("crm", "v2")
pkgv3 := makeTestLocalPackageReference("crm", "v3")
```

This will allow you to write the test cases more compactly:

``` go
		{
			left: MakeTypeName( v2, "TypeA"),
			right: MakeTypeName{v3, "TypeB"),
			expected: true,
		},
```

Reply:
Kindly review